analyze/analyze.rs

```diff
@@ -276,6 +276,8 @@ pub fn dominators(
 
 struct Paths {
     items: Vec<ir::Id>,
+    max_depth: usize,
+    max_row: usize,
```

Comment:
Rather than add a copy of each field from the options, and add another field here every time we add an option, we should include a copy of the options here.

```
opts: opt::Paths
```